### ISSUE TITLE
function for deleting channel

### ISSUE THREAD

**Issue by kansi** [cabol/erlbus#25]
I tried to look around in the code base but couldn't find any function for deleting a channel. Did I miss it or is it not there ?

**Comment by cabol** [comment]
You're right, in this momento `erlbus` doesn't implement `delete` function for channels. This is because the original idea was to have idempotent operations, so if you take a look at `ebus_pg2` if you subscribe to a channel and it doesn't exist, then that channel is created and then the subscription, and the same with the other operations -- is something similar what `gproc` does. But I agree that probably could be useful to have explicitly `delete` function in some cases. I'll consider your suggestion as a new feature, so this will remain as an issue. Meanwhile if you already have an idea or something implemented, or any other suggestion/comment, please feel free to share it!

Thanks!

**Comment by kansi** [comment]
Since you granted me the liberty to give my suggestions I will point out a few things:
- currently the context can't be updated by the handler. I would really wish the context to be dynamically updatable and a quick look at the code makes me feel that it is possible to do so. You would need to just accept return values here https://github.com/cabol/erlbus/blob/master/src/ebus_handler.erl#L170 and https://github.com/cabol/erlbus/blob/master/src/ebus_worker.erl#L75 and then update the context accordingly.
- antoher thing I stumbled upon while checking the feasible of the above was the way poolboy is being initalized and the way it is being used :
  initilization: https://github.com/cabol/erlbus/blob/master/src/ebus_handler.erl#L117-L119
  usage : https://github.com/cabol/erlbus/blob/master/src/ebus_handler.erl#L165
  
    Since you are using cast to send messages poolboy will send the message to the worker and then return the worker to the list of available workers. Also poolboy by default uses lifo stratergy so it will keep choosing this returned worker again and again rather than picking the next worker. So you will need to change the stratergy to fifo if you want the incomming messages to evenly distributed among the workers which i guess is the whole point of having workers.

**Comment by cabol** [comment]
Good suggestions, I agree with them. So we'll work on them soon!

Thanks!!!
